packages/plugin-client-drizzle/test/integration.ava.ts

```diff
@@ -0,0 +1,2187 @@
+  t.deepEqual(res, [
+    {
+      users2: {
+        id: 1,
+        name: 'John',
+        cityId
+      },
+      cities: {
+        id: cityId,
+        name: 'Paris',
+        state: null
+      }
+    },
+    {
+      users2: {
+        id: 2,
+        name: 'Jane',
+        cityId: null
+      },
+      cities: null
+    }
+  ]);
+});
+
+test.skip('join subquery', async (t) => {
+  const { db } = t.context;
+
+  await db
+    .insert(courseCategoriesTable)
+    .values([{ name: 'Category 1' }, { name: 'Category 2' }, { name: 'Category 3' }, { name: 'Category 4' }]);
+
+  await db.insert(coursesTable).values([
+    { name: 'Development', categoryId: 2 },
+    { name: 'IT & Software', categoryId: 3 },
+    { name: 'Marketing', categoryId: 4 },
+    { name: 'Design', categoryId: 1 }
+  ]);
+
+  const sq2 = db
+    .select({
+      categoryId: courseCategoriesTable.id,
+      category: courseCategoriesTable.name,
+      total: sql<number>`count(${courseCategoriesTable.id})`
+    })
+    .from(courseCategoriesTable)
+    .groupBy(courseCategoriesTable.id, courseCategoriesTable.name)
+    .as('sq2');
+
+  const res = await db
+    .select({
+      courseName: coursesTable.name,
+      categoryId: sq2.categoryId
+    })
+    .from(coursesTable)
+    .leftJoin(sq2, eq(coursesTable.categoryId, sq2.categoryId))
+    .orderBy(coursesTable.name);
+
+  t.deepEqual(res, [
+    { courseName: 'Design', categoryId: 1 },
+    { courseName: 'Development', categoryId: 2 },
+    { courseName: 'IT & Software', categoryId: 3 },
+    { courseName: 'Marketing', categoryId: 4 }
+  ]);
+});
+
+test.skip('with ... select', async (t) => {
+  const { db } = t.context;
+
+  await db.insert(orders).values([
+    { region: 'Europe', product: 'A', amount: 10, quantity: 1 },
+    { region: 'Europe', product: 'A', amount: 20, quantity: 2 },
+    { region: 'Europe', product: 'B', amount: 20, quantity: 2 },
+    { region: 'Europe', product: 'B', amount: 30, quantity: 3 },
+    { region: 'US', product: 'A', amount: 30, quantity: 3 },
+    { region: 'US', product: 'A', amount: 40, quantity: 4 },
+    { region: 'US', product: 'B', amount: 40, quantity: 4 },
+    { region: 'US', product: 'B', amount: 50, quantity: 5 }
+  ]);
+
+  const regionalSales = db.$with('regional_sales').as(
+    db
+      .select({
+        region: orders.region,
+        totalSales: sql<number>`sum(${orders.amount})`.as('total_sales')
+      })
+      .from(orders)
+      .groupBy(orders.region)
+  );
+
+  const topRegions = db.$with('top_regions').as(
+    db
+      .select({
+        region: regionalSales.region
+      })
+      .from(regionalSales)
+      .where(
+        gt(regionalSales.totalSales, db.select({ sales: sql`sum(${regionalSales.totalSales})/10` }).from(regionalSales))
+      )
+  );
+
+  const result = await db
+    .with(regionalSales, topRegions)
+    .select({
+      region: orders.region,
+      product: orders.product,
+      productUnits: sql<number>`sum(${orders.quantity})::int`,
```

Comment:
It seems to me that aliasing is missing the query. We run `sum("quantity")::int, sum("amount")::int`, and `productUnits` and `productSales` is missing from the query.
So I guess it has to be adjusted in the adapter to make sure the sums are displayed with the correct name. Or we can alias the `sum` with the name in the query.